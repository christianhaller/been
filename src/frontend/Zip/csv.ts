import json2csv from "json2csv/dist/json2csv.umd";
import type { EnhancedPin } from "../../backend/interace";

export function csv(data: EnhancedPin[]) {
  const fields = ["lat", "lon", "country", "city", "been"];
  return json2csv.parse(
    data.map(({ flags, lng, lat, country, city }) => {
      return {
        been: flags.join(","),
        lon: lng,
        lat,
        country,
        city,
      };
    }),
    {
      delimiter: ";",
      quote: "",
      fields,
    }
  );
}
