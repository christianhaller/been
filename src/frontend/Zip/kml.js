import tokml from "tokml";
import GeoJSON from "geojson";
export function kml(data) {
    const json = GeoJSON.parse(data.places, {
        Point: ["lat", "lng"],
    });
    return tokml(json, {
        name: data.username + "'s travelamp map",
        documentName: "tbd",
        documentDescription: "tbd",
    });
}
