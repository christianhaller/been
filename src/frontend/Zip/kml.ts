import tokml from "tokml";
import GeoJSON from "geojson";
import type { Response } from "../../backend/interace";

export function kml(data: Response) {

  const json = GeoJSON.parse(data.places.map((place)=>{
    return {
      ...place,
      name: `${place.country}, ${place.city}`
    }
  }), {
    Point: ["lat", "lng"],
    include:['name','flags']
  });  
  return tokml(json, {
    name: 'name',
    documentName: `${data.username}'s travelmap`,
    documentDescription: `I have been to ${data.places.length} places`,
  });
}
