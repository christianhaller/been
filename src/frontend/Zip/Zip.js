var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import JSZip from "jszip/dist/jszip";
import { kml } from "./kml";
import { csv } from "./csv";
export class Zip {
    create(data, flags, png) {
        return __awaiter(this, void 0, void 0, function* () {
            const { username } = data;
            const zip = new JSZip();
            zip.file(`${username}.csv`, csv(data.places));
            zip.file(`${username}.kml`, kml(data));
            zip.file(`${username}.png`, png);
            if (flags) {
                zip.file(`${username}.txt`, flags);
            }
            return zip.generateAsync({ type: "blob" });
        });
    }
}
