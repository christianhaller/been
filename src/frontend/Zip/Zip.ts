import JSZip from "jszip/dist/jszip";
import { kml } from "./kml";
import { csv } from "./csv";
import type { Response } from "../../backend/interace";

export class Zip {
  async create(
    data: Response,
    flags: string | undefined,
    png: Blob
  ): Promise<Blob> {
    const { username } = data;
    const zip = new JSZip();

    zip.file(`${username}.csv`, csv(data.places));
    zip.file(`${username}.kml`, kml(data));
    zip.file(`${username}.png`, png);
    if (flags) {
      zip.file(`${username}.txt`, flags);
    }
    return zip.generateAsync({ type: "blob" });
  }
}
