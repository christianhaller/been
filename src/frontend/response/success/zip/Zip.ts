import { downloadZip } from "client-zip";
import { kml } from "./kml";
import { csv } from "./csv";
import type { Response } from "../../../../backend/interace";

export class Zip {
  async create(
    data: Response,
    flags: string | undefined,
    png: Blob
  ): Promise<Blob> {
    const { username } = data;

    const files = [
      ["kml", kml(data)],
      ["csv", csv(data.places)],
      [
        "txt",
        `${
          flags || ""
        } wanna say thanks? https://www.paypal.com/paypalme/christianhaller/2`,
      ],
      ["png", png],
    ].map(([type, input]) => {
      return {
        name: `${username}.${type}`,
        input,
      };
    });
    return downloadZip(files).blob();
  }
}
