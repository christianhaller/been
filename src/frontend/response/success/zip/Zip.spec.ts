import { Zip } from "./Zip";
import type { Response } from "../../../../backend/interace";

jest.mock("client-zip", () => ({
  downloadZip: jest.fn(() => {
    return {
      blob: () => {
        return Promise.resolve(new Blob());
      },
    };
  }),
}));

describe("Zip", () => {
  test("create", async () => {
    const res = new Zip().create(
      {
        language: "en",
        username: "christianhaller",
        places: [
          {
            lat: 50.444885,
            lng: 30.536667,
            flags: ["been"],
            city: "Kyiv(Kiev)",
            country: "Ukraine",
          },
          {
            lat: -34.45451,
            lng: -57.83755,
            flags: ["been"],
            city: "Colonia del Sacramento",
            country: "Uruguay",
          },
        ],
      } as Response,
      "ðŸ‡²ðŸ‡°",
      new Blob([])
    );

    await expect(res).resolves.toMatchSnapshot();
  });
});
