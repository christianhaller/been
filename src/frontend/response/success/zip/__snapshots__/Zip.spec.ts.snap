// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Zip create 1`] = `Blob {}`;
